ref/haskell/src/Lib.hs

```diff
@@ -0,0 +1,114 @@
+{-# LANGUAGE OverloadedStrings #-}
```

Comment:
I did that, and the overloaded strings were especially nice in the Spec.hs (unit tests). I am not sure though if it is worth it. As I understand it, OverloadedStrings advanced fast to one of the most used extensions. Not sure which other compilers there that do not implement this extension, but the others that ByteString etc. rely on. (BangPatterns, for example).